src/web/src/pages/CoursePage.vue

```diff
@@ -0,0 +1,87 @@
+<template>
+  <div v-if="ready" class="container w-100 mb-4">
+    <b-row>
+      <b-col>
+        <h1 class="mt-4">{{ courseTitle }}</h1>
+        <h4 class="mb-1">{{ courseName }}</h4>
+      </b-col>
+    </b-row>
+    <b-row>
+      <b-col>
+        {{
+          generateRequirementsText(
+            courseObj.prerequisites,
+            courseObj.corequisites,
+            courseObj.raw_precoreqs
+          )
+        }}
+      </b-col>
+    </b-row>
+    <b-row>
+      <b-col class="mb-4">
+        <br />
+        {{ courseObj.description }}
+      </b-col>
+    </b-row>
+    <b-button :to="backRoute">Back</b-button>
+  </div>
+  <div v-else>
+    <b-spinner></b-spinner>
+    <strong class="m-2">Loading Course...</strong>
+  </div>
+</template>
+
+<script>
+import { getCourses } from "../services/YacsService";
+import { getDefaultSemester } from "@/services/AdminService";
+import { generateRequirementsText } from "@/utils";
+
+export default {
+  name: "CoursePage",
+  data() {
+    return {
+      ready: false,
+      courseName: this.$route.params.course,
+      courseTitle: String,
+      courseObj: {},
+      allCoursePreReqs: {},
+      selectedSemester: String,
+      backRoute: String,
+    };
+  },
+  methods: {
+    generateRequirementsText,
+  },
+  async created() {
+    const querySemester = this.$route.query.semester;
+    this.selectedSemester =
+      querySemester && querySemester != "null"
+        ? querySemester
+        : await getDefaultSemester();
+    getCourses(this.selectedSemester).then((courses) => {
+      for (const c of courses) {
+        this.allCoursePreReqs[c.name] = c.prerequisites;
```

Comment:
This should be relegated to backend, probably  a `/course/:id` route, but this is fine for now